test/distributed/fsdp/test_fsdp_checkpoint.py

```diff
@@ -310,6 +312,77 @@ def forward(self, x):
         return self.l2(self.relu(self.s2(self.s1(self.l1(x)))))
 
 
+# FSDP(
+# X(
+# A() # no checkpoint
+# Checkpoint(Block())
+# C() # no checkpoint
+# )
+# )
+
+
+class Block(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.lin = nn.Linear(10, 10)
+
+    def forward(self, x):
+        return self.lin(x)
+
+
+class BlockHolder(nn.Module):
+    def __init__(self, rank, checkpoint_indices, reentrant):
+        super().__init__()
+        torch.manual_seed(rank)
```

Comment:
It is not clear to me why we want to initialize the model to be different on each rank.